src/bootstrap/src/utils/shared_helpers.rs

```diff
@@ -13,8 +13,9 @@ use std::io::Write;
 use std::process::Command;
 use std::str::FromStr;
 
-#[cfg(test)]
-mod tests;
+// If we were to declare a tests submodule here, the shim binaries that include this
+// module via `#[path]` would fail to find it, which breaks `./x check bootstrap`.
+// So instead the unit tests for this module are in `super::tests::shared_helpers_tests`.
```

Comment:
Remark: FTR, this seems a bit nonsense that a unit test can't be next to or inside the unit, but anyway

Reply:
In other parts of the repo changing a test would force a complete recompile unless the test is placed in a separate file that's gated on `cfg(test)`. This is less of an issue for bootstrap since recompiling isn't that slow (and regularly happens anyway).

Reply:
I considered adding an exception to tidy, but the workaround was easy enough, and isn't really any dirtier than adding an exception would have been.